### PR TITLE
CDAP-7482 Fix system log backward compatibility

### PR DESCRIPTION
**Pull request by rohitsinha54**
Tested on cluster. 

TODO:
- [x]  Add a long running test for to test reading system logs

### CODE REVIEW

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java [hunk 1]
```diff
@@ -366,4 +375,22 @@ private static String getValueFromTag(LoggingContext.SystemTag tag) {
...
+    // .TAG_NAMESPACE_ID does not exist we use ServiceLoggingContext.TAG_SYSTEM_ID to support backward
+    // compatibility for the logs which are in kafka and needs to be written to HDFS through log saver. See CDAP-7482
+    return tags.containsKey(NamespaceLoggingContext.TAG_NAMESPACE_ID) ? tags.get
+      (NamespaceLoggingContext.TAG_NAMESPACE_ID) : tags.get(ServiceLoggingContext.TAG_SYSTEM_ID);
+  }
+
+  @Nullable
+  private static String getNamespaceOrSystemIDFromSystemTagMap(Map<String, LoggingContext.SystemTag> systemTagsMap) {
+    Map<String, String> newMap = new HashMap<>();
+    for (Map.Entry<String, LoggingContext.SystemTag> entry : systemTagsMap.entrySet()) {
```
Comment (chtyim) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java]: Better use Maps transform to avoid copying

**Comment by poornachandra** [comment]
LGTM

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java [hunk 2]
```diff
@@ -366,4 +378,13 @@ private static String getValueFromTag(LoggingContext.SystemTag tag) {
                 context.getSystemTagsMap().get(ServiceLoggingContext.TAG_SERVICE_ID).getValue());
     return builder.build();
   }
+
+  @Nullable
+  private static <T> T getNamespaceOrSystemID(Map<String, T> tags) {
```
Comment (chtyim) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java]: Should be call `getByNamespaceOrSystemID`.

File: cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java [hunk 3]
```diff
@@ -211,14 +209,28 @@ public static LoggingContext getLoggingContext(String namespaceId, String applic
 
   public static Filter createFilter(LoggingContext loggingContext) {
     if (loggingContext instanceof ServiceLoggingContext) {
-      String systemId = loggingContext.getSystemTagsMap().get(ServiceLoggingContext.TAG_NAMESPACE_ID).getValue();
+      LoggingContext.SystemTag namespaceOrSystemID = getNamespaceOrSystemID(loggingContext.getSystemTagsMap());
+      if (namespaceOrSystemID == null) {
+        throw new IllegalArgumentException("No namespace or system id present");
+      }
+      String systemId = namespaceOrSystemID.getValue();
```
Comment (chtyim) [inline cdap-watchdog/src/main/java/co/cask/cdap/logging/context/LoggingContextHelper.java]: Seems confusing after calling `getValue()`, the `namespaceOrSystemID` becomes `systemId`. Better call them `systemTag` and `systemId`.

**Comment by rohitsinha54** [comment]
Build: http://builds.cask.co/browse/CDAP-RUT251-2
IT: http://builds.cask.co/browse/CDAP-ITM0-44

**Comment by rohitsinha54** [comment]
Tested on cluster across upgrade from 3.4 to 3.5 and also tested with the log running tests for system logs: https://github.com/caskdata/cdap-integration-tests/pull/468
Waiting on build to pass. IT has already passed.

**Comment by rohitsinha54** [comment]
Build passed. Merging.

**Pull request merged**